pool.go

```diff
@@ -1,30 +1,47 @@
 package rueidis
 
-import "sync"
+import (
+	"sync"
+	"time"
+)
 
-func newPool(cap int, dead wire, makeFn func() wire) *pool {
+func newPool(cap int, dead wire, idleConnTTL *time.Duration, minSize int, makeFn func() wire) *pool {
 	if cap <= 0 {
 		cap = DefaultPoolSize
 	}
 
-	return &pool{
-		size: 0,
-		cap:  cap,
-		dead: dead,
-		make: makeFn,
-		list: make([]wire, 0, 4),
-		cond: sync.NewCond(&sync.Mutex{}),
+	p := &pool{
+		size:         0,
+		minSize:      minSize,
+		cap:          cap,
+		dead:         dead,
+		make:         makeFn,
+		list:         make([]wire, 0, 4),
+		cond:         sync.NewCond(&sync.Mutex{}),
+		idleConnTTL:  idleConnTTL,
+		lastUsage:    make(map[wire]time.Time),
+		stopChan:     make(chan struct{}, 1),
+		stopChanOnce: &sync.Once{},
 	}
+
+	go p.removeIdleConns()
+
+	return p
 }
 
 type pool struct {
-	dead wire
-	cond *sync.Cond
-	make func() wire
-	list []wire
-	size int
-	cap  int
-	down bool
+	dead         wire
+	cond         *sync.Cond
+	make         func() wire
+	list         []wire
+	size         int
+	minSize      int
+	cap          int
+	down         bool
+	idleConnTTL  *time.Duration
+	stopChan     chan struct{}
+	stopChanOnce *sync.Once
```

Comment:
```suggestion
	timer  *time.Timer
```
Can we just use `time.AfterFunc` and store the timer here? I think a timer is convenient for us because we can use `.Stop()` when `len(p.list) <= minSize` and `.Reset()` when `len(p.list) > minSize`.